Lib/threading.py

```diff
@@ -925,7 +930,7 @@ def run(self):
         """
         try:
             if self._target is not None:
-                self._target(*self._args, **self._kwargs)
+                self._context.run(self._target, *self._args, **self._kwargs)
```

Comment:
That's one why to fix the issue. Another would be to modify the contextvars module to detect that it runs in a new thread. For exemple, this change doesn't cover the case of threads spawned by C libraries which then use the Python C API to run Python code. Or functions calling _thread.start_new_thread() directly.

I don't know which option is the best. At least, this change is reasonable small.

Reply:
Sorry, maybe I add bad description for this issue. I will try to make things more clear. Every thread has it's own context and they do not share same context. This PR tries fix problem when child thread use empty context instead of copy of a parent context.